packages/client/src/links/httpSubscriptionLink.ts

```diff
@@ -35,25 +38,45 @@ async function urlWithConnectionParams(
   return url;
 }
 
-type HTTPSubscriptionLinkOptions<TRoot extends AnyClientTypes> = {
+type HTTPSubscriptionLinkOptions<
+  TRoot extends AnyClientTypes,
+  TEventSource extends EventSourceLike.AnyConstructor = typeof EventSource,
+> = {
   /**
-   * EventSource options or a callback that returns them
+   * EventSource ponyfill
    */
-  eventSourceOptions?: CallbackOrValue<EventSourceInit>;
+  EventSource?: TEventSource;
```

Comment:
Love that you've done this, I was thinking to do the same, should make it all much easier

Do we need a link to the docs in the jsdoc comment?